beacon-chain/blockchain/forkchoice/service.go

```diff
@@ -41,18 +41,20 @@ type Store struct {
 	checkpointStateLock  sync.Mutex
 	seenAtts             map[[32]byte]bool
 	seenAttsLock         sync.Mutex
+	slasherClient        ethpb.SlasherClient
 }
 
 // NewForkChoiceService instantiates a new service instance that will
 // be registered into a running beacon node.
-func NewForkChoiceService(ctx context.Context, db db.Database) *Store {
+func NewForkChoiceService(ctx context.Context, db db.Database, sc ethpb.SlasherClient) *Store {
 	ctx, cancel := context.WithCancel(ctx)
 	return &Store{
 		ctx:             ctx,
 		cancel:          cancel,
 		db:              db,
 		checkpointState: cache.NewCheckpointStateCache(),
 		seenAtts:        make(map[[32]byte]bool),
+		slasherClient:   sc,
```

Comment:
Same here as feeding slasher client to fork choice service. It feels like a missuse of responsibility